### DISCUSSION TITLE
App companion non comunica

### DISCUSSION THREAD

**Discussion by Dim-1** [indomus/forum#1742] [Networking]
Salve a tutti, forse mi sto perdendo in un bicchier d'acqua data la mia inesperienza.
Veniamo al problema: sul mio telefono Oppo A9 Android ho installato la app di Home assistant. Tutto funziona quando sono nella stessa wifi di casa (dove è presente anche il miniPC con su homeassistant); quando esco di casa ovviamente  la app non si connette più al server homeassistant, in quanto non ho ancora attivato nessun servizio per collegarmi da remoto (DuckDNS). Il problema è che non viene fatto nessun tracciamento del device, ne vengono aggiornati i vari sensori del telefono fino a che non rientro nella wifi domestica. Sul telefono ho autorizzato tutto, funzionamento in background, tolto risparmio energatico, sbloccato sicurezze, eccetera. Ma niente, il mio telefono viene visto sempre a casa, non avvengono tracciamenti. Suggerimenti?

**Comment by Giancky79** [comment]
Sicuramente i cinesi hanno preso il controllo del chip che fa funzionare il gps, rendendo vano qualsiasi modifica della posizione, so con certezza che se attivi un dominio e usi duckdns se vai di hassos o caddy se vai con core, questo blocca lo script cinese e permette di avere il gps di nuovo attivo, prova segrndo le guide di indomus e vedrai che risolvi

**Comment by maxcanna** [comment]
> la app non si connette più al server homeassistant, in quanto non ho ancora attivato nessun servizio per collegarmi da remoto (DuckDNS)

non pensi che sia questa la causa? se non si connette ovviamente 

> non viene fatto nessun tracciamento del device, ne vengono aggiornati i vari sensori del telefono

Come dovrebbe fare l'app a comunicare con HA?

**Comment by Dim-1** [comment]
Ho ricercato su vari siti, letto, eccetera, e non ho mai trovato scritto che, per aggiornare i sensori del telefono, si dovesse essere collegati direttamente in remoto ad HA con duckdns o simili. Per quello che conosco io del backend di home assistant(cioè niente), i dati dei vari sensori del telefono potevano arrivare al server anche in altro modo (in fondo un cloud è presente). Non ho mai avuto pretesa di collegarmi all'interfaccia grafica da remoto senza aver installato qualche componente. Nelle tante guide lette, nessuna ha mai detto che, per fare aggiornare i sensori, servisse duckdns come prerequisito. Evidentemente mi sbagliavo di brutto, perdonatemi se offendo la vostra grande intelligenza.

**Reply by KC-inDomus** [reply]
ma scusa, è una questione di buon senso, pensaci: in qualche modo 'sti benedetti sensori "a qualcosa" dovranno esser comunicati, nello specifico alla tua istanza home assistant. se non è raggiungibile da remoto, come ci arrivano, i dati.

**Reply by NdR91** [reply]
Ma perdonami. 
La tua app, ora, sa solo che la tua istanza di HA è raggiungibile tramite un IP che sarà qualcosa tipo "192.168.1....", quindi un indirizzo LOCALE,

Non sa altro.

Come può, all'esterno della tua rete, raggiungere un IP locale per condividere i dati dei sensori?? 

Non è una questione di interfaccia.

Il discorso Cloud, perdonami, non ha senso. Cosa ti aspetteresti da questo "cloud" che dici essere presente?

**Reply by NdR91** [reply]
Tra l'altro, aggiungo, un servizio Cloud esiste ma è a pagamento.

**Reply by snakuzzo** [reply]
Il cloud ? Installi l'app sul cell e funziona tutto, giusto ? Così...senza fare niente :D

**Reply by Dim-1** [reply]
Ho capito, è che speravo che lo spirito Santo mi aiutasse🤷🏻‍♂️😊

**Reply by indomus** [reply]
Configura HA in modo che sia accedibile da remoto in modo (il più possibile) sicuro. 👍
